openhands/cli/tui.py

```diff
@@ -700,20 +700,207 @@ def _(event: KeyPressEvent) -> None:
+                escaped_part = '[^\\s]+\\.[a-zA-Z0-9]+'
+                break
+
+        # Handle special cases like .py (starting with dot) - but only if it's a standalone extension
+        if (
+            escaped_part.startswith('\\.')
+            and len(escaped_part) > 2
+            and '/' not in escaped_part
+        ):
+            # Check if this looks like a file extension pattern (e.g., .py, .txt)
+            ext_part = escaped_part[2:]  # Remove the \.
+            if ext_part.isalpha() and len(ext_part) <= 5:  # Common extension length
+                escaped_part = '\\.[a-zA-Z0-9]+'
+
+        # Replace numbers with flexible number patterns
+        escaped_part = re.sub(r'\\\d+', r'\\d+', escaped_part)
+
+        pattern_parts.append(escaped_part)
+
+    # Join with flexible whitespace
+    pattern = '\\s+'.join(pattern_parts)
+
+    # Make the pattern match from start to end
+    return f'^{pattern}$'
+
+
+def _parse_piped_command(command: str) -> list[str]:
+    """Parse a command that may contain pipes into individual commands.
+
+    Args:
+        command: The command string to parse.
+
+    Returns:
+        list[str]: List of individual commands split by pipes.
+    """
+    import shlex
+
+    # Handle edge cases first
+    if not command or not command.strip():
+        return []
+
+    # Check for invalid pipe patterns
+    if command.strip().startswith('|') or command.strip().endswith('|'):
+        return []  # Invalid: starts or ends with pipe
+
+    if '||' in command:
+        return []  # Invalid: double pipes
+
+    # Split by pipe operator, handling quoted strings properly
+    parts = []
+    current_part: list[str] = []
+
+    # Use shlex to properly handle quoted strings and escapes
+    try:
+        tokens = shlex.split(command, posix=True)
+    except ValueError:
+        # If shlex fails (e.g., unmatched quotes), fall back to simple split
+        tokens = command.split()
+
+    # Reconstruct the command while splitting on pipes
+    i = 0
+    while i < len(tokens):
+        token = tokens[i]
+        if token == '|':
+            if current_part:
+                parts.append(' '.join(current_part))
+                current_part = []
+            else:
+                # Empty part before pipe - invalid
+                return []
+        else:
+            current_part.append(token)
+        i += 1
+
+    # Add the last part if it exists
+    if current_part:
+        parts.append(' '.join(current_part))
+    elif parts:  # We had pipes but no final command
+        return []  # Invalid: ends with pipe
+
+    # If no pipes were found, fall back to simple pipe splitting
+    if len(parts) <= 1 and '|' in command:
+        simple_parts = [part.strip() for part in command.split('|')]
+        # Check for empty parts
+        if any(not part for part in simple_parts):
+            return []  # Invalid: empty parts
+        parts = simple_parts
```

Comment:
@openhands we don't need to fall back, in case there are actual command that include `|` in part of the string argument. please simply this function